src/internal/kinesis.ts

```diff
@@ -0,0 +1,490 @@
+            // Also, note the inconsistency in casing...
+            const errorMessage: string =
+                (error.Message as string) || (error.message as string) || (error.__type as string)
+
+            // Handle specifically the case of an invalid signature
+            if (error.__type === 'InvalidSignatureException') {
+                throw new InvalidSignatureError(errorMessage, error.__type)
+            }
+
+            // Otherwise throw a standard service error
+            throw new KinesisServiceError(errorMessage, error.__type as string, operation)
+        }
+
+        if (errorCode === 1500) {
+            throw new KinesisServiceError(
+                'An error occured on the server side',
+                'InternalServiceError',
+                operation
+            )
+        }
+    }
+
+
+
+    /**
+     * It creates a stream.
+     * @param {CreateStreamRequest} request - CreateStreamRequest - The request object that will be
+     * sent to the server.
+     */
+    createStream(request: CreateStreamRequest): void {
+        this.RequestOperation(`CreateStream`, request)
+    }
+
+    /**
+     * It deletes a stream.
+     * @param request - Partial<DeleteStreamRequest>
+     */
+    deleteStream(request: Partial<DeleteStreamRequest>): void {
+        this.RequestOperation(`DeleteStream`, request)
+    }
+
+    /**
+     * It returns a list of streams.
+     * @param request - Partial<ListStreamRequest> = {}
+     * @returns A partial of the ListStreamsResponse class.
+     */
+    listStreams(request: Partial<ListStreamRequest> = {}): Partial<ListStreamsResponse> {
+        const res = this.RequestOperation(`ListStreams`, request)
+        return ListStreamsResponse.fromJson(res.json())
+    }
+
+    /**
+     * It takes a request object and returns a response object.
+     * @param {PutRecordsRequest} request - PutRecordsRequest
+     * @returns A partial of the PutRecordsResponse class.
+     */
+    putRecords(request: PutRecordsRequest): Partial<PutRecordsResponse> {
+        const res = this.RequestOperation(`PutRecords`, request)
+        return PutRecordsResponse.fromJson(res.json())
+    }
+
+    /**
+     * It gets the records from the database.
+     * @param {GetRecordsRequest} request - GetRecordsRequest
+     * @returns A partial of the GetRecordsResponse class.
+     */
+    getRecords(request: GetRecordsRequest): Partial<GetRecordsResponse> {
+        const res = this.RequestOperation(`GetRecords`, request)
+        return GetRecordsResponse.fromJson(res.json())
+    }
+
+    /**
+     * It lists the shards in a stream.
+     * @param {ListShardsRequest} request - The request object that is passed to the API.
+     * @returns ListShardsResponse
+     */
+    listShards(request: ListShardsRequest): ListShardsResponse {
+        const res = this.RequestOperation(`ListShards`, request)
+        return ListShardsResponse.fromJson(res.json())
+    }
+
+
+    /**
+     * 
```

Comment:
```suggestion
     * Gets an Amazon Kinesis shard iterator. A shard iterator expires 5 minutes after it is returned to the requester.
     *
```

Reply:
I'm not familiar with Kinesis: this returns an "iterator", which seems to be some id by the looks of it. Do we have any methods of making any use of it, though? Otherwise, we'd drop this method altogether. Once again I'm not familiar with Kinesis, so, looking forward to your 💡 here 🙏🏻